lib/auth/auth.go

```diff
@@ -923,7 +958,118 @@ func (a *Server) runPeriodicOperations() {
 			a.syncReleaseAlerts(ctx, true)
 		case <-localReleaseCheck.Next():
 			a.syncReleaseAlerts(ctx, false)
+		case <-instancePeriodics.Next():
+			// instance periodics are rate-limited and may be time-consuming in large
+			// clusters, so launch them in the background.
+			go a.doInstancePeriodics(ctx)
+		}
+	}
+}
+
+func (a *Server) doInstancePeriodics(ctx context.Context) {
+	const slowRate = time.Millisecond * 200 // 5 reads per second
+	const fastRate = time.Millisecond * 5   // 200 reads per second
+	const dynamicPeriod = time.Minute * 3
+
+	instances := a.GetInstances(ctx, types.InstanceFilter{})
+
+	// dynamically scale the rate-limiting we apply to reading instances
+	// s.t. we read at a progressively faster rate as we observe larger
+	// connected instance counts. this isn't a perfect metric, but it errs
+	// on the side of slowness, which is preferable for this kind of periodic.
+	instanceRate := slowRate
+	if ci := a.inventory.ConnectedInstances(); ci > 0 {
+		localDynamicRate := dynamicPeriod / time.Duration(ci)
+		if localDynamicRate < fastRate {
+			localDynamicRate = fastRate
 		}
+
+		if localDynamicRate < instanceRate {
+			instanceRate = localDynamicRate
+		}
+	}
+
+	limiter := rate.NewLimiter(rate.Every(instanceRate), 100)
+	instances = stream.RateLimit(instances, func() error {
+		return limiter.Wait(ctx)
+	})
+
+	// cloud deployments shouldn't include control-plane elements in
+	// metrics since information about them is not actionable and may
+	// produce misleading/confusing results.
+	var skipControlPlane bool
+	if modules.GetModules().Features().Cloud {
+		skipControlPlane = true
+	}
```

Comment:
This can be expressed in a much simpler form.

```suggestion
	skipControlPlane := modules.GetModules().Features().Cloud
```